### PR TITLE
Create component.

### PR DESCRIPTION
**Pull request by xerona**
https://jira.inbcu.com/browse/AUG-614

### CODE REVIEW

File: src/select-search.component.ts [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+    );
+    this.renderer.setElementStyle(
+      this.dropdownContents.nativeElement,
+      'max-width',
+      `${this.dropdownWidth}px`
+    );
+
+    this.active = !this.active;
+    if (this.active) {
+      this.ref.detectChanges();
```
Comment (bcandullo) [inline src/select-search.component.ts]: Is this needed? Looks like ref is attached and CDS is default.

Comment (xerona) [inline src/select-search.component.ts]: That, in conjunction with `doLayout()` prevents a little visual ugliness. With that in place, the entire dropdown contents is populated when it appears. Without it, on a few rows appear at first, then the rest are rendered later.

**Comment by paulkahle** [comment]
Hey @xerona 
This looks really good.  A couple of enhancement ideas came to me when I was using it.  I think they should be straight forward but let me know:
1. Include an option for setting the search box placeholder text
2. Make the debounce setting for the search box configurable
3. Include the search icon in the search box . (ClearableInputComponent.search=true)
4. Add an ability to focus the search box when the dropdown opens.  Make this configurable.
Thanks.

**Comment by xerona** [comment]
@paulkahle This is ready now

**Comment by paulkahle** [comment]
@xerona 
The selected value needs to be an Input property so the value can be set from outside the component.

**Comment by paulkahle** [comment]
@xerona 
I've found a bug by which the value emitted in the valueChanged event is the previously selected value, not the value just chosen.

**Comment by xerona** [comment]
Thanks! I missed that, but I'll update it now.

**Comment by paulkahle** [comment]
@xerona 
One more thing.  Can the selected item be highlighted in the dropdown?

**Comment by xerona** [comment]
@paulkahle I added the missing highlight

**Pull request merged**